lib/editor-linter.coffee

```diff
@@ -35,6 +33,10 @@ class EditorLinter
   onDidDestroy: (callback) ->
     return @emitter.on('did-destroy', callback)
 
+  destroy: ->
+    @emitter.emit('did-destroy')
```

Comment:
It's actually called after TextEditor is destroyed, not before it, here we are emulating the same event, to support the `linter:toggle` command and emulate a TextEditor destruction.